File: Data Structures & Algorithms/depth-of-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root, int currDepth) {
        currDepth++;

        int left = currDepth;
        int right = currDepth;

        if(root.left != null){
            left = maxDepth(root.left,currDepth);
        }

        if(root.right != null){
            right = maxDepth(root.right,currDepth);
        }

        return right - left >= 0 ? right: left;


    }
    
    public int maxDepth(TreeNode root) {
        if(root == null){
            return 0;
        }

        int currDepth = 1;
        int left = currDepth;
        int right = currDepth;

        if(root.left != null){
            left = maxDepth(root.left,currDepth);
        }

        if(root.right != null){
            right = maxDepth(root.right,currDepth);
        }

        return right - left >= 0 ? right: left;
        
        
    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        self.res = 0
        def helper(root: TreeNode) -> int:
            if root is None:
                return 0
            
            left = helper(root.left)
            right = helper(root.right)

            m = max(left,right)
            self.res = max(self.res, m, left +right)
            return m+1

        helper(root)
        return self.res